portfolio/src/main/webapp/calendarApi.js

```diff
@@ -0,0 +1,298 @@
+
+  // Fetches the user's Calendar's time zone. 
+  gapi.client.calendar.settings.get({'setting': 'timezone'})
+      .then((responseTimeZone) => {
+        const userCalendarTimeZone =
+            encodeURIComponent(responseTimeZone.result.value);
+        const calendarViewUrl =
+            'https://calendar.google.com/calendar/embed?src=' + userEmail +
+            '&ctz=' + userCalendarTimeZone + '&mode=WEEK';
+
+        // Updates the UI so that calendar view appears.
+        $('#calendar-view').removeClass('d-none');
+        $('#calendar-view').attr('src', calendarViewUrl);
+        $('#calendar-view').show();
+      });
+}
+
+/**
+ * Onclick function for 'Looks good to me, export" button. 
+ * Asks the user for Write access to the API scope.
+ */
+function addWriteScope() {
+  hideErrorMessages();
+  var GoogleUser = GoogleAuth.currentUser.get();
+  GoogleUser.grant({'scope': SCOPE_READ_WRITE})
+      .then((response) => {
+        addNewEventsToGoogleCalendar();
+      })
+      .catch(function(error) {
+        handleExportAuthError(error);
+      })
+}
+
+/**
+ * Adds the scheduled task items back to the user's Google Calendar.
+ * TODO(hollyyuqizheng): change the hard-coded part later when scheduled 
+ * results are returned.
+ */
+function addNewEventsToGoogleCalendar() { 
+  const now = new Date();
+  const fiveHoursLater = new Date();
+  fiveHoursLater.setHours(now.getHours() + 5);
+
+  const event = {
+    'summary': 'Testing',
+    'description': 'test description',
+    'start': {
+      'dateTime': now.toISOString(),
+    },
+    'end': {
+      'dateTime': fiveHoursLater.toISOString(),
+    },
+  };
+
+  addOneEventToGoogleCalendar(event);
+}
+
+/** Adds an individual event to the authorized user's Google Calendar. */
+function addOneEventToGoogleCalendar(event) {
+  const request = gapi.client.calendar.events.insert(
+      {'calendarId': 'primary', 'resource': event});
+
+  request.execute(function() {
+    // Refreshes the calendar view so that the new event shows up on it.
+    showCalendarView(GoogleAuth.currentUser.get());
+  });
+}
+
+/**
+ * Updates import message box based on the error during authentication process
+ * for exporting
+ */
+function handleExportAuthError(e) {
+  $('#export-calendar-message').removeClass('d-none');
+  if (e.error === ERROR_CODE_POPUP_CLOSED) {
+    $('#export-calendar-message')
+        .text(
+            'It seems like you didn\'t complete the authorization process. ' +
+            'Please click the Export button again.');
+  } else if (e.error === ERROR_CODE_ACCESS_DENIED) {
+    $('#export-calendar-message')
+        .text(
+            'You didn\'t give permission to update your Google Calendar, ' +
+            'so your calendar events cannot be exported.');
+  }
+  $('#export-calendar-message').show();
+}
+
+/** Hides the error messages when authentication completes succesfully. */
+function hideErrorMessages() {
+  $('#import-calendar-message').hide();
```

Comment:
I'd suggest that you consistently rely on the d-none class name instead of using show() and hide().

Reply:
Updated.